wordpress-mobile/WordPress-FluxC-Android
# [Migrate] Kotlin Android Extensions to ViewBinding

Depends On:
 - [x] [ADC#32](https://github.com/Automattic/android-dependency-catalog/pull/32)
 - [x] Update ADC to `1.17.0`

### Description

This PR is about migrating from the deprecated `kotlin-android-extensions` plugin, replacing Kotlin synthetics with ViewBinding and using `kotlin-parcelize` instead.

FYI: The `kotlin-android-extensions` plugin has been deprecated for years now and it is now preventing us from updating to a newer Kotlin version (ie. [Kotlin 1.9.25](https://github.com/JetBrains/kotlin/releases/tag/v1.9.25)).

### To Test

- Smoke test the `FluxC Example` app, try out all available screens and functionality. Verify everything is working as expected.
- Also, if you want to be thorough about reviewing the `kotlinx.parcelize.Parcelize` import change, you could quickly smoke test, either the [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android) and/or [WCAndroid](https://github.com/woocommerce/woocommerce-android), with this version of `FluxC`, or via composite builds, and see if it works as expected.
